File: Lab 5/TPyramid.py

from TFigure import TFigure
import math

class TPyramid(TFigure):
    def __init__(self, height, seclen, faces):
        super().__init__(height)
        self.__faces = faces
        self.__seclen = seclen

    @property
    def faces(self):
        return self.__faces

    @faces.setter
    def faces(self, faces):
        self.__faces = faces

    @property
    def seclen(self):
        return self.__seclen

    @seclen.setter
    def seclen(self, seclen):
        self.__seclen = seclen

    def GetV(self) ->float:
        return self.GetSOs()*self._height/3
    
    def GetS(self) -> float:
        return self.GetApof() * self.GetPerimetrOs() / 2 * self.__faces + self.GetSOs()

    def GetSOs(self)-> float:
        return self.GetOsApof() * self.GetPerimetrOs() / 2

    def GetOsApof(self)-> float:
        return self.__seclen / (2 * math.tan(math.pi / self.__faces))

    def GetPerimetrOs(self)-> float:
        return self.__faces * self.__seclen

    def GetApof(self)-> float:
        return math.sqrt(self.GetOsApof() ** 2 + self._height ** 2)

File: Lab 5/TPyramid.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_5
{
    internal class TPyramid : TFigure,ICloneable
    {
        int faces;
        double seclen;
        public TPyramid(double height ,int faces, double seclen):base(height)
        { 
            this.faces = faces;
            this.seclen = seclen;            
        }
        public int Faces { get { return faces; } set { faces = value; } }
        public double Seclen { get { return seclen; } set { seclen = value; } }
        public override double GetS()
        {
            return GetApof()*GetPerimetrOs()/2*faces + GetSOS();
        }

        public override double GetV()
        {
            return GetSOS()*height/3;
        }

        protected override double GetSOS()
        {
            return (GetOsApof() * GetPerimetrOs())/2;
        }

        double GetOsApof()
        {
            return seclen / (2*Math.Tan(Math.PI/faces));
        }

        double GetPerimetrOs()
        {
            return faces * seclen;
        }

        double GetApof()
        {
            return Math.Sqrt(Math.Pow(GetOsApof(), 2)+height*height);
        }

        public object Clone()
        {
            return new TPyramid(this.height,this.faces, this.seclen);
        }
    }
}
